src/cli/index.js

```diff
@@ -339,7 +339,9 @@ function onUnexpectedError(err: Error) {
   log.push(`Trace: ${indent(err.stack)}`);
 
   const errorLoc = path.join(config.cwd, 'yarn-error.log');
-  fs.writeFileSync(errorLoc, log.join('\n\n') + '\n');
+  try {
+    fs.writeFileSync(errorLoc, log.join('\n\n') + '\n');
+  } catch (err) {}
```

Comment:
The thing is, [#346](https://github.com/yarnpkg/yarn/pull/2095/files/8ffa389e437d7a2f3aaedb7804765a131c53bed9#diff-867becf4a9c2c6c6d4e7c1278750724eR346) already handles it marvelously. It just needs to get there. I was thinking to rather piggyback on that one, rather than handling it explicitly. But I can of course change into doing so.